### PR TITLE
replace bash subcommands with separate shell spawns

### LINKED ISSUE #54: Some warnings
Just noting a few warnings I got while preparing the v9.9.0 release:

```console
$ NODE_OPTIONS="--trace-warnings" changelog-maker --group --start-ref v9.8.0
(node:10896) [DEP0005] DeprecationWarning: The Buffer() and new Buffer() constructors are not recommended for use due to security and usability concerns. Please use the new Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() construction methods instead.
    at showFlaggedDeprecation (buffer.js:139:13)
    at new Buffer (buffer.js:160:3)
    at BufferList.copy (/home/mzasso/.nvm/versions/node/v9.8.0/lib/node_modules/changelog-maker/node_modules/gitexec/node_modules/bl/bl.js:108:19)
    at BufferList.slice (/home/mzasso/.nvm/versions/node/v9.8.0/lib/node_modules/changelog-maker/node_modules/gitexec/node_modules/bl/bl.js:99:15)
    at BufferList.end (/home/mzasso/.nvm/versions/node/v9.8.0/lib/node_modules/changelog-maker/node_modules/gitexec/node_modules/bl/bl.js:89:31)
    at Socket.onend (_stream_readable.js:602:10)
    at Object.onceWrapper (events.js:272:13)
    at Socket.emit (events.js:185:15)
    at endReadableNT (_stream_readable.js:1101:12)
    at process._tickCallback (internal/process/next_tick.js:114:19)
Your GitHub username: targos
Your GitHub password: XXXXXX

(node:10896) TimeoutOverflowWarning: 4294967296000 does not fit into a 32-bit signed integer.
Timer duration was truncated to 2147483647.
    at Object.exports.enroll (timers.js:407:13)
    at TLSSocket.Socket.setTimeout (net.js:369:12)
    at TLSSocket.<anonymous> (_http_client.js:725:14)
    at Object.onceWrapper (events.js:272:13)
    at TLSSocket.emit (events.js:185:15)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1157:10)
Your GitHub OTP/2FA Code (optional): XXXXXX
(node:10896) TimeoutOverflowWarning: 4294967296000 does not fit into a 32-bit signed integer.
Timer duration was truncated to 2147483647.
    at Object.exports.enroll (timers.js:407:13)
    at TLSSocket.Socket.setTimeout (net.js:369:12)
    at TLSSocket.<anonymous> (_http_client.js:725:14)
    at Object.onceWrapper (events.js:272:13)
    at TLSSocket.emit (events.js:185:15)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1157:10)
```

### LINKED ISSUE #63: Make it work on Windows
Not sure how much work will this be, but currently this doesn't work on Windows.

I'll add more info later.

### PR DESCRIPTION
**Pull request by rvagg**
As per discussion in #66

I need to think about this a bit more, there's likely going to be overhead for non-windows users because this replaces all Bash subcommands with separate `spawn(shell)` executions and piecing the output together to form the final `git log` command.

It also leaves this repo in a 1/2 promisified state which is a bit awkward.

But, let's see what CI does.

Closes #63, fixes #54

### CODE REVIEW

**Comment by rvagg** [comment]
:thumbsup: passing on windows, though I've removed 8.x in the latest commit, `stream.pipeline()` is a 10.x feature and `util.promisify()` is improved in 10.x too.

**Comment by XhmikosR** [comment]
Remember that there's a case where things worked on Windows CI but still failed on vanilla Windows cmd :)

Please ping me when you think this is ready so that I try it on my Windows VM.

**Comment by XhmikosR** [comment]
@rvagg just tested this on my Windows VM, with `/origin/xmr-pack-gitexec-up` merged in this branch and seems to work fine.

BTW, so that we are safe maybe we should close #66 and you rebase this branch to include the patch instead?

File: package.json [hunk 1]
```diff
@@ -7,7 +7,6 @@
     "changelog-maker": "./changelog-maker.js"
   },
   "author": "Rod <rod@vagg.org> (https://r.va.gg/)",
-  "license": "MIT",
```
Comment (XhmikosR) [inline package.json]: Why was this removed?

Comment (rvagg) [reply package.json]: yeah, good question! I have no idea, but I'll rectify.

**Comment by rvagg** [comment]
I can measure a very small performance impact by this. In the order of 10-20ms, and it's going to depend a bit on I/O speed.

I think that having this work on Windows is probably worth the price. Unwinding some of the complexity of the git commands is also going to be a bonus here I think--the way it's currently structured makes it quite unfriendly for contributors I think. I would like some other opinions on that though.

This change will be semver-major because we're dropping support for older Node. I wouldn't mind getting at least a new `ghauth()` in before a release too, that should mostly take care of the TODO in here but will be a separate PR because it needs a refactor of `collect-commit-labels.js`.

**Comment by XhmikosR** [comment]
Yup, I noticed a small perf too. But I agree with your points.
